iosApp/Survey/Sources/Presentation/Modules/Splash/SplashView.swift

```diff
@@ -0,0 +1,61 @@
+//
+//  SplashView.swift
+//  Survey
+//
+//  Created by Bliss on 26/9/22.
+//  Copyright © 2022 Nimble. All rights reserved.
+//
+
+import FlowStacks
+import SwiftUI
+
+struct SplashView: View {
+
+    @State var animating = false
+
+    let showLogin: () -> Void
```

Comment:
So we will send back to `AppCoordinator` to handle about navigation?

How about this?
```swift
struct App: View {

	@State var routes: Routes<Screen> = [.root(.splash)]

    var body: some View {
        Router($routes) { screen, _ in
            switch screen {
            case .splash:
                SplashView()
            case .login:
                LoginView()
            }
        }
		.environmentObject(routes)
    }
}

// then
struct SplashView: View {
	@EnvironmentObject var routes: Routes<Screen>

	...
	routes = [.root(.login)]
	...
}
```

Reply:
Good idea 😄 One issue I have with this approach is that we have to expose all functions from the Router and the Router itself which I think is not safe. Because `EnvironmentObject` does not allow protocol programming.  I adapted your suggestion in https://github.com/blyscuit/kmm-swiftUI-IC/pull/55/commits/ba7a44e89315166d5395bb7a07b2fecbad91fd7a.